community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/state/Loaders.java

```diff
@@ -145,9 +145,9 @@ public void ensureHeavy( RelationshipRecord record )
             }
 
             @Override
-            public RelationshipRecord clone(RelationshipRecord relationshipRecord) {
-                // Not needed because we don't manage before state for relationship records.
-                throw new UnsupportedOperationException("Unexpected call to clone on a relationshipRecord");
+            public RelationshipRecord clone(RelationshipRecord relationshipRecord)
+            {
+                return relationshipRecord.clone();
```

Comment:
Looks like `Loader#clone()` is not needed now because all implementations simply delegate to the actual record.